asg.tf

```diff
@@ -1,6 +1,6 @@
 # Create launch configration
 resource "aws_launch_configuration" "launch_configuration" {
-  name                 = "${var.tags["Environment"]}-${var.tags["Application"]}-${var.tags["Tier"]}-${var.name}-LC"
+  name_prefix          = "${var.tags["Environment"]}-${var.tags["Application"]}-${var.tags["Tier"]}-${var.name}-LC-"
```

Comment:
Hyphenation is the standard for the tagging policy - it uses no underscores, just forwasrd slashes and hyphens.

Perhaps a forwardslash?

"${var.tags["Environment"]}-${var.tags["Application"]}-${var.tags["Tier"]}-${var.name}-LC/"